src/base/static/context/language.js

```diff
@@ -0,0 +1,25 @@
+import React, { createContext, Component } from "react";
+import PropTypes from "prop-types";
+
+const LanguageContext = createContext();
+
+export const LanguageConsumer = LanguageContext.Consumer;
+
+export class LanguageProvider extends Component {
+  render() {
+    return (
+      <LanguageContext.Provider
+        value={{
+          currentLanguage: this.props.currentLanguage,
+          defaultLanguage: this.props.defaultLanguage,
+        }}
```

Comment:
React Context docs warn against using object literals in the Context's `value` prop: https://reactjs.org/docs/context.html#caveats

As suggested there, we should life this up into component state.

Also, I believe we should be able to make this a functional component using hooks. I think it would be worth giving it a try here, with the `useState` hook.

Reply:
I think the better approach will be to leverage the existing context provided by i18next, rather than duplicate a lot of what's already provided. So I decided to completely remove our `Language` context. RIP context API...